src/models/event-timeline-set.ts

```diff
@@ -702,6 +702,28 @@ export class EventTimelineSet extends TypedEventEmitter<EmittedEvents, EventTime
             );
         }
 
+        if (timeline.getTimelineSet() !== this) {
+            throw new Error(`EventTimelineSet.addEventToTimeline: Timeline=${timeline.toString()} does not belong " + 
+                "in timelineSet(threadId=${this.thread?.id})`);
+        }
```

Comment:
I'm confused by this SonarQube code coverage analysis, https://sonarcloud.io/component_measures?id=matrix-js-sdk&metric=new_coverage&pullRequest=2856&view=list

We have tests for both of these new conditions.

Reply:
Latest results: https://sonarcloud.io/component_measures?id=matrix-js-sdk&metric=new_coverage&pullRequest=2856&view=list

Reply:
The cause was a left-in focused test `fit` causing the others not to be run, see https://github.com/matrix-org/matrix-js-sdk/pull/2856#discussion_r1015415938